File: Data Structures & Algorithms/house-robber/submission-3.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    rob(nums: number[]): number {
        const dp = []

        if (nums.length == 1) {
            return nums[0]
        } else if (nums.length == 2) {
            return nums[0] > nums[1] ? nums[0] : nums[1]
        }

        dp.push(nums[0])
        dp.push(nums[1] > nums[0] ? nums[1] : nums[0])
        for (let i = 2 ; i < nums.length ; i++) {
            let max = Math.max(nums[i] + dp[i-2], dp[i-1])
            dp.push(max)
        }
        return dp.pop()
    }
}

File: Data Structures & Algorithms/binary-search/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1

        while l <= r :
            i = (l + r) // 2
            # print(i)
            if target == nums[i]:
                return i
            if target < nums[i]:
                r = i-1
            elif target > nums[i]:
                l = i+1

        return -1


        